packages/core/src/__tests__/embrace.test.ts

```diff
@@ -1,204 +1,228 @@
+      addSessionProperty: (key: string, value: string, permanent: boolean) =>
+        mockAddSessionProperty(key, value, permanent),
+      removeSessionProperty: (key: string) => mockRemoveSessionProperty(key),
+      setUserAsPayer: () => mockSetUserAsPayer(),
+      clearUserAsPayer: () => mockClearUserAsPayer(),
+      setJavaScriptBundlePath: (path: string) =>
+        mockSetJavaScriptBundlePath(path),
+      logNetworkRequest: (
+        url: string,
+        httpMethod: MethodType,
+        startInMillis: number,
+        endInMillis: number,
+        bytesSent: number,
+        bytesReceived: number,
+        statusCode: number,
+        error: string,
+      ) =>
+        mockLogNetworkRequest(
+          url,
+          httpMethod,
+          startInMillis,
+          endInMillis,
+          bytesSent,
+          bytesReceived,
+          statusCode,
+          error,
+        ),
+      logNetworkClientError: (
+        url: string,
+        httpMethod: MethodType,
+        startInMillis: number,
+        endInMillis: number,
+        errorType: string,
+        errorMessage: string,
+      ) =>
+        mockLogNetworkClientError(
+          url,
+          httpMethod,
+          startInMillis,
+          endInMillis,
+          errorType,
+          errorMessage,
+        ),
+      getLastRunEndState: () => mockGetLastRunEndState(),
+      getDeviceId: () => mockGetDeviceId(),
+      getCurrentSessionId: () => mockGetCurrentSessionId(),
+    },
+  },
+}));
+
+const mockSt = "this is a fake stack trace";
+const mockGenerateStackTrace = jest.fn();
+jest.mock("../utils/ErrorUtil", () => ({
+  ...jest.requireActual("../utils/ErrorUtil"),
+  generateStackTrace: () => mockGenerateStackTrace(),
+}));
 
 describe("User Identifier Tests", () => {
+  const testUserId = "Lucia";
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
   test("setUserIdentifier", async () => {
-    const mock = jest.fn();
-    jest.mock(
-      "react-native",
-      () => ({
-        NativeModules: {
-          EmbraceManager: {
-            setUserIdentifier: mock,
-          },
-        },
-      }),
-      {virtual: true},
-    );
-    const {setUserIdentifier} = require("../index");
     await setUserIdentifier(testUserId);
-    expect(mock).toHaveBeenCalledWith(testUserId);
+    expect(mockSetUserIdentifier).toHaveBeenCalledWith(testUserId);
   });
 
   test("clearUserIdentifier", async () => {
-    const mock = jest.fn();
-    jest.mock(
-      "react-native",
-      () => ({
-        NativeModules: {
-          EmbraceManager: {
-            clearUserIdentifier: mock,
-          },
-        },
-      }),
-      {virtual: true},
-    );
-
-    const {clearUserIdentifier} = require("../index");
     await clearUserIdentifier();
-    expect(mock).toHaveBeenCalled();
+    expect(mockClearUserIdentifier).toHaveBeenCalled();
   });
 });
 
 describe("User Data Tests", () => {
+  const testUserId = "Lucia";
+  const testEmail = "lucia@nimble.la";
```

Comment:
I know you are not supposed to refactor, but maybe it worth to change this to something like test@test.com?

Reply:
yep makes sense, updated